pkg/reconciler/accessor/istio/virtualservice.go

```diff
@@ -74,8 +75,8 @@ func ReconcileVirtualService(ctx context.Context, owner kmeta.Accessor, desired
 		// Don't modify the informers copy
 		existing := vs.DeepCopy()
 		existing.Spec = desired.Spec
-		existing.Labels = desired.Labels
-		existing.Annotations = desired.Annotations
+		existing.Labels = resources.UnionMaps(vs.Labels, desired.Labels)
+		existing.Annotations = resources.UnionMaps(vs.Annotations, desired.Annotations)
```

Comment:
This means we will copy down added/modified annotations, but not remove deleted annotations.  Seems odd.

Reply:
Yes, it seems odd... And I realized that we are not allowed to add annotation to `routes` actually.

```
$ kubectl annotate routes.serving.knative.dev hello-example foo=bar
route.serving.knative.dev/hello-example annotated

$ kubectl get routes.serving.knative.dev hello-example -o jsonpath='{.metadata.annotations}'
map[serving.knative.dev/creator:admin serving.knative.dev/lastModifier:admin]
```

We don't need this PR and the Ingress and VirtualService should have same behavior?